File: java/src/main/java/api/cornell/data/classes/Bool.java
package api.cornell.data.classes;

/**
 * A simple {@code Bool} enum for Cornell APIs.
 */
public enum Bool {
    /**
     * YES.
     */
    Y,
    /**
     * NO.
     */
    N;
    
    /**
     * Converts this value to an equivalent boolean value.
     *
     * @return the equivalent boolean value.
     */
    public boolean toBoolean() {
        switch (this) {
            case Y:
                return true;
            case N:
                return false;
            default:
                throw new RuntimeException("Impossible!");
        }
    }
}

File: kotlin/src/main/kotlin/api/cornell/data/classes/Bool.kt
package api.cornell.data.classes

/**
 * A simple [Bool] enum for Cornell APIs.
 */
enum class Bool {
    Y, N;

    /**
     * [toBoolean] converts the [Bool] value to a boolean.
     */
    fun toBoolean() = when (this) {
        Y -> true
        N -> false
    }
}
